docs/conf.py

```diff
@@ -517,6 +521,17 @@ def _add_jinja_filters(app):
         app.builder.templates.environment.filters["env_override"] = env_override
 
 
+def install_mermaid(
+    app: Sphinx,
+    pagename: str,
+    templatename: str,
+    context: Dict,
+    doctree: Optional[nodes.document],
+) -> None:
+    if doctree and doctree.next_node(mermaid):
+        app.add_js_file("https://unpkg.com/mermaid/dist/mermaid.js")
+
+
```

Comment:
Does this only add it to pages that have `{mermaid}`?! :o

Reply:
Yep! But it still isn't working properly, and it's driving me insane...

I've also tried creating a local copy of the `mermaid.js` (or `mermaid.min.js`) file under `_static`, and it loads fine, but doesn't render. I think it does work if I just `app.add_js_file("js/mermaid.min.js")` without this conditional handler, and it adds to all the files.

I'm going to try reverting back to adding to all files, and then taking the approach of removing the script from all pages that don't have a mermaid node. My current hypothesis is that the rendering of the text that get's converted to a Mermaid diagram is messed up without this, but really not sure...